src/new-client/src/services/executions.ts

```diff
@@ -154,6 +160,26 @@ const executionsMap$ = newExecution$.pipe(
           return concat(
             execute(payload).pipe(
               map((response) => mapResponseToExecution(response, execution.id)),
+              takeUntil(timer(REQEUST_TIMEOUT)),
+              publish((request) =>
+                merge(
+                  request,
+                  timer(CLIENT_TIMEOUT).pipe(
+                    mapTo({
+                      ...execution,
+                      status: ExecutionStatus.ClientTimeout,
+                    }),
+                    takeUntil(request),
+                  ),
+                  timer(REQEUST_TIMEOUT).pipe(
+                    mapTo({
+                      ...execution,
+                      status: ExecutionStatus.RequestTimeout,
+                    }),
+                    takeUntil(request),
+                  ),
+                ),
+              ),
```

Comment:
I find this a bit difficult to follow/understand. The fact that the timer for the `REQUEST_TIMEOUT` is defined twice is a bit confusing IMO.

What I suggest is that:

- We improve the `execute` function so that:
  - it takes the `Execution` as its parameter
  - it also maps the response
  - it races that ^ stream with a timer that will map to a RequestTimeout execution

That way the stream returned by `execute` will handle all the logic/side-effects related with the execution itself.

- Then, since it's likely that we will want to apply the behavior of emitting a value if the original stream hasn't emitted in a certain time, let's create a piepable operator named `withTooLong` message that has the following signature:
```ts
const withTooLongMessage: <M, T>(ms: number, message: M) => OperatorFunction<T, M | T>
```
For that you can use the technique described in this [this blog-post](https://ncjamieson.com/calling-publish-with-a-selector/#with-a-selector)

- After having done that, then we just need to put it together:
```ts
exhaustMap((execution) =>                              
  concat(                                              
    execute(execution).pipe(                           
      withTooLongMessage(TAKING_TOO_LONG, {             
        ...execution,                                  
        status: ExecutionStatus.ClientTimeout,         
      }),                                              
    ),                                                 
    race([                                             
      executionDismiss$.pipe(                          
        filter((x) => x.currencyPair === currencyPair),
        take(1),                                       
      ),                                               
      timer(DISMISS_TIMEOUT),                          
    ]).pipe(mapTo(blankExecution(currencyPair))),      
  ),                                                   
),                                                     
```